barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -580,17 +578,16 @@ template <typename Builder> cycle_group<Builder> cycle_group<Builder>::operator+
         // Constrain x_diff * lambda = y2 - y1
         field_t::evaluate_polynomial_identity(x_diff, lambda, -y2, y1);
     }
-    const field_t x3 = lambda.madd(lambda, -(x2 + x1)); // x3 = lambda^2 - x1 - x2
-    const field_t y3 = lambda.madd(x1 - x3, -y1);       // y3 = lambda * (x1 - x3) - y1
-    cycle_group add_result(x3, y3, /*is_infinity=*/x_coordinates_match);
```

Comment:
Here (and in the subtract method) I'm removing construction of an actual `cycle_group` element for the add/sub result because (1) only the coordinates are being used anyway and (2) this would trigger the on-curve assert failure during normal operations because the result is by design _not_ on the curve if `x_coordinates_match == true`